### PR TITLE
Add filter for bytecode generated by Compose Kotlin compiler plugin

### LINKED ISSUE #1208: Jacoco reports low coverage with Compose tests
### Steps to reproduce

* JaCoCo version:  0.8.7
* Operating system: macOS Big Sur 11.4
* Tool integration: Gradle
* Complete executable reproducer: https://github.com/arcadefire/JacocoComposeSample
* Steps: 
  - Checkout to the repo above and execute the command: `./gradlew clean JacocoTestReport`

### Expected behaviour
Creating a simple composable function displaying a text and a test covering it that checks that the text was displayed should produce 100% test coverage.

### Actual behaviour
The coverage for such a simple case is particularly low, especially the _conditional_ coverage, even though the function contains no conditional logic.

Screenshot 2021-07-04 at 21 00 02

Also, looking at the function's coverage details, the yellow diamonds report a number of untested conditional branches.

Screenshot 2021-07-04 at 21 00 55

### PR DESCRIPTION
**Pull request by Godin**
There is no validation test because there is no official way to use Kotlin Compose Compiler Plugin from Maven.

However here is [example.zip](https://github.com/jacoco/jacoco/files/15195058/example.zip) for which execution of

```
./gradlew build jacocoTestReport
```

before this change produces

before1

before2

and with this change installed into local Maven repository produces

after1

after2

Also, I propose to create a repository based on this example with Cl build that will act as a validation test.

Fixes #1208
Closes #1259

### CODE REVIEW

File: org.jacoco.core.test/src/org/jacoco/core/internal/analysis/filter/KotlinComposeFilterTest.java [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+		final Range range2 = new Range();
+		final Range range3 = new Range();
+		final Range range4 = new Range();
+		final Range range5 = new Range();
+		final Range range6 = new Range();
+		final Range range7 = new Range();
+
+		m.visitVarInsn(ALOAD, 1);
+		range1.fromInclusive = m.instructions.getLast();
+		m.visitLdcInsn(new Integer(-974630231));
```
Comment (marchof) [inline org.jacoco.core.test/src/org/jacoco/core/internal/analysis/filter/KotlinComposeFilterTest.java:95]:
```suggestion
		m.visitLdcInsn(Integer.valueOf(-974630231));
```
Avoid deprecation warning on newer JDKs.

File: org.jacoco.core.test/src/org/jacoco/core/internal/analysis/filter/KotlinComposeFilterTest.java [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+		m.visitJumpInsn(IFNE, label6);
+		range1.toInclusive = m.instructions.getLast();
+
+		m.visitLabel(label5);
+		m.visitMethodInsn(INVOKESTATIC, "androidx/compose/runtime/ComposerKt",
+				"isTraceInProgress", "()Z", false);
+		Label label7 = new Label();
+		m.visitJumpInsn(IFEQ, label7);
+		range3.fromInclusive = m.instructions.getLast();
+		m.visitLdcInsn(new Integer(-974630231));
```
Comment (marchof) [inline org.jacoco.core.test/src/org/jacoco/core/internal/analysis/filter/KotlinComposeFilterTest.java:144]:
```suggestion
		m.visitLdcInsn(Integer.valueOf(-974630231));
```
Avoid deprecation warning on newer JDKs.

**Review by marchof — CHANGES_REQUESTED** [review]
There are two compiler warnings in the unit test on modern JDKs that could be fixed.

**Review by marchof — APPROVED** [review]

**Comment by marchof** [comment]
Looks like the Azure pipeline had a network failure:

    Caused by: org.apache.maven.wagon.TransferFailedException: GET request of: org/codehaus/plexus/plexus-utils/3.5.1/plexus-utils-3.5.1.jar from central failed
        at org.apache.maven.wagon.AbstractWagon.getTransfer (AbstractWagon.java:413)

**Comment by Godin** [comment]
Passed after restart 😉

File: org.jacoco.core.test/src/org/jacoco/core/internal/analysis/filter/KotlinComposeFilterTest.java (hunk 1 above)
Comment (Godin) [reply org.jacoco.core.test/src/org/jacoco/core/internal/analysis/filter/KotlinComposeFilterTest.java]: BTW this code was generated by ASMifier, so this fix of warning can be contributed to it -  https://gitlab.ow2.org/asm/asm/-/blob/ASM_9_7/asm-util/src/main/java/org/objectweb/asm/util/ASMifier.java#L1483 😉

**Comment by Godin** [comment]
> Also, I propose to create a repository based on this example with Cl build that will act as a validation test.

@marchof I assume that silence gives consent - https://github.com/jacoco/example-jacoco-kotlin-compose 😉

**Comment by marchof** [comment]
Sorry, I overlooked this comment. Sure, thanks for creating the repo.

**Comment by Abdallah-Abdelazim** [comment]
@Godin This is a very important fix for Android projects using Jacoco. Can we please release this to the stable version?

**Comment by Godin** [comment]
See FAQ https://www.jacoco.org/jacoco/trunk/doc/faq.html

> When will feature X be released?
>
> JaCoCo is maintained by volunteers in their free time. Since we cannot guarantee free capacity, we do not commit to particular release dates. Typically, you can expect a couple of releases every year.
>
> In the [change log](https://www.jacoco.org/jacoco/trunk/doc/changes.html), you can see all features that have been implemented in master branch and will be available with the next release. And **in the meantime you can test latest build of master branch** ([Maven SNAPSHOT](https://www.jacoco.org/jacoco/trunk/doc/repo.html)) and provide [feedback](https://www.jacoco.org/jacoco/trunk/doc/support.html) to us.

**Comment by szczepanski-k** [comment]
@Godin Thank you for this fix! Do we have any ETA for release 0.8.13?

File: org.jacoco.core.test/src/org/jacoco/core/internal/analysis/filter/KotlinComposeFilterTest.java (hunk 1 above)
Comment (Godin) [reply org.jacoco.core.test/src/org/jacoco/core/internal/analysis/filter/KotlinComposeFilterTest.java]: And here is PR for ASM: https://gitlab.ow2.org/asm/asm/-/merge_requests/422

**Pull request merged**